config/config.go

```diff
@@ -17,23 +17,25 @@ const (
 var EmptyNRWrapper = "Undefined"
 
 type Configuration struct {
-	ExtensionEnabled   bool
-	LicenseKey         string
-	LicenseKeySecretId string
-	NRHandler          string
-	TelemetryEndpoint  string
-	LogEndpoint        string
-	RipeMillis         uint32
-	RotMillis          uint32
-	LogLevel           string
-	SendFunctionLogs   bool
-	LogServerHost      string
+	ExtensionEnabled        bool
+	LicenseKey              string
+	LicenseKeySecretId      string
+	LicenseKeyParameterName string
+	NRHandler               string
+	TelemetryEndpoint       string
+	LogEndpoint             string
+	RipeMillis              uint32
+	RotMillis               uint32
+	LogLevel                string
+	SendFunctionLogs        bool
+	LogServerHost           string
 }
 
 func ConfigurationFromEnvironment() *Configuration {
 	enabledStr, extensionEnabledOverride := os.LookupEnv("NEW_RELIC_LAMBDA_EXTENSION_ENABLED")
 	licenseKey, lkOverride := os.LookupEnv("NEW_RELIC_LICENSE_KEY")
 	licenseKeySecretId, lkSecretOverride := os.LookupEnv("NEW_RELIC_LICENSE_KEY_SECRET")
+	licenseKeyParameterName, lkParameterOverride := os.LookupEnv("NEW_RELIC_LICENSE_KEY_PARAMETER")
```

Comment:
"PARAMETER" is really generic. It has a specific meaning here, it terms of the SSM Agent. Let's find a name that makes the context clear.